modules/workspace.nix

```diff
@@ -23,14 +23,26 @@ let
     };
   };
 
+  desktopIconSortingModeId = {
+    manual = -1;
+    name   = 0;
+    size   = 1;
+    date   = 2;
+    type   = 6;
+  }.${cfg.workspace.desktop.icons.sorting.mode};
```

Comment:
And drop `.${cfg.workspace.desktop.icons.sorting.mode}` at the end. You can convert the value in an apply statement for the option, i.e.:
```nix
sorting = {
          mode = lib.mkOption {
            type = with lib.types; nullOr (enum [ "manual" "name" "size" "type" "date" ]);
            default = null;
            example = "type";
            description = ''
              Specifies the sort mode for the desktop icons. By default they are
              sorted by name.
            '';
           apply = sortMode: if (sortMode == null) then null else desktopIconSortingModeId.${sortMode};
          };
```

Reply:
Cool, I haven’t encountered `apply` before. I couldn’t find a lot of documentation I could understand so I don’t really know how it works, but [these][1] [two][2] links make it look like the result of the `apply` function will be the actual value of the option. If that is the case, should I also change [this line][3] like that?

```diff
-${stringIfNotNull cfg.workspace.desktop.icons.sorting.mode ''desktop.writeConfig("sortMode", ${builtins.toString desktopIconSortingModeId});''}
+${stringIfNotNull cfg.workspace.desktop.icons.sorting.mode ''desktop.writeConfig("sortMode", ${builtins.toString cfg.workspace.desktop.icons.sorting.mode});''}
```

[1]: https://ryantm.github.io/nixpkgs/functions/library/options/#function-library-lib.options.mkOption
[2]: https://github.com/NixOS/nixpkgs/issues/16884#issuecomment-238814281
[3]: https://github.com/nix-community/plasma-manager/blob/0a076cdcaf6b718b18b41f929cfa49d658f9f2f7/modules/workspace.nix#L424

Reply:
Exactly like this!